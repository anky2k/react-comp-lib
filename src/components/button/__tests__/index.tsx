import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Button from '../../button';

afterEach(cleanup);

test('it should render correctly with outlined true option', () => {
  const clickHandler = jest.fn();
  const { getByText } = render(
    <Button outlined onClick={clickHandler} label="Submit" />
  );
  expect(getByText('Submit')).toBeInTheDocument();
  fireEvent.click(getByText('Submit'));
  expect(clickHandler).toHaveBeenCalled();
});

test('it should render correctly with outlined false option', () => {
  const clickHandler = jest.fn();
  const { getByText } = render(
    <Button onClick={clickHandler} label="Submit" />
  );

  expect(getByText('Submit')).toBeInTheDocument();

  fireEvent.click(getByText('Submit'));
  expect(clickHandler).toHaveBeenCalled();
});

test('it should render correctly without label', () => {
  const clickHandler = jest.fn();
  const { getByText } = render(<Button onClick={clickHandler} />);

  expect(getByText('Some label')).toBeInTheDocument();

  fireEvent.click(getByText('Some label'));
  expect(clickHandler).toHaveBeenCalled();
});

test('it should render as per given styles', () => {
  const clickHandler = jest.fn();

  const { asFragment } = render(
    <Button onClick={clickHandler} label="Submit" />
  );

  expect(asFragment()).toMatchSnapshot();
});
