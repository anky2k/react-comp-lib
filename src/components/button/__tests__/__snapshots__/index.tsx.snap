// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render as per given styles 1`] = `
<DocumentFragment>
  <button
    class="rounded outline-none shadow py-3 px-12 font-normal uppercase tracking-wider text-lg bg-purple-600 border border-purple-600 text-white"
  >
    <span>
      Submit
    </span>
  </button>
</DocumentFragment>
`;
